ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/extensions/PaywallDataExtensions.kt

```diff
@@ -0,0 +1,102 @@
+package com.revenuecat.purchases.ui.revenuecatui.extensions
+
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import androidx.compose.ui.platform.LocalContext
+import androidx.core.os.LocaleListCompat
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.paywalls.PaywallColor
+import com.revenuecat.purchases.paywalls.PaywallData
+import com.revenuecat.purchases.ui.revenuecatui.data.processed.PaywallTemplate
+import java.net.URL
+
+/***
+ * Default [PaywallData] to display when attempting to present a [PaywallView] with an offering that has no paywall
+ * configuration, or when that configuration is invalid.
+ */
+@Composable
+@ReadOnlyComposable
+internal fun PaywallData.Companion.createDefault(packages: List<Package>): PaywallData {
+    return PaywallData.createDefaultForIdentifiers(packages.map { it.identifier })
+}
+
+@Composable
+@ReadOnlyComposable
+private fun PaywallData.Companion.createDefaultForIdentifiers(packageIdentifiers: List<String>): PaywallData {
+    val context = LocalContext.current
+    val locale = LocaleListCompat.getDefault()[0].toString()
+
+    return PaywallData(
+        templateName = PaywallData.defaultTemplate,
+        config = PaywallData.Configuration(
+            packages = packageIdentifiers,
+            images = PaywallData.Configuration.Images(
+                background = PaywallData.defaultBackgroundImage,
+                icon = PaywallData.defaultAppIconPlaceholder,
+            ),
+            colors = PaywallData.defaultColors,
+            blurredBackgroundImage = true,
+            displayRestorePurchases = true,
+        ),
+        localization = mapOf(locale to PaywallData.defaultLocalization), // TODO-PAYWALLS: test this
+        assetBaseURL = PaywallData.defaultTemplateBaseURL(context.packageName),
+        revision = PaywallData.revisionID,
+    )
+}
+
+// region Private defaults
+
+private val PaywallData.Companion.defaultTemplate: String // TODO-PAYWALLS: use enum
+    get() = PaywallTemplate.TEMPLATE_2.id
+
+private val PaywallData.Companion.defaultAppIconPlaceholder: String // TODO-PAYWALLS: use real icon
```

Comment:
Done right?

Reply:
Hmm I think this image doesn't exist? I think we need to have a little code to access the app's icon. But we should be able to with the context.

Reply:
Yeah, it doesn't exist. But there's also no usages of icons in Template 1 and 2 so I left this like this for now

Reply:
I made some changes to this. Made it display the app icon if the icon is equal to this placeholder. I tested it and it looks fine. The only thing is that it shows round, since the app icon is a round icon